src/components/PostDetails.tsx

```diff
@@ -1,106 +1,124 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { createComment, deleteComment, getPostComments } from './api/users';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isErrorComments, setIsErrorComments] = useState(false);
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const isNoComments =
+    selectedPost && comments.length === 0 && !isLoading && !isErrorComments;
+
+  useEffect(() => {
+    setIsErrorComments(false);
+    setIsShownForm(false);
+    if (selectedPost) {
+      setIsLoading(true);
+      getPostComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setIsErrorComments(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedPost]);
+
+  const addComment = ({ name, email, body, postId }: Omit<Comment, 'id'>) => {
+    return createComment({ name, email, body, postId }).then(newComment =>
+      setComments(currentComments => [...currentComments, newComment]),
+    );
+  };
```

Comment:
You are using `setComments` right after `createComment`. It's not the best approach because `createComment` is asynchronous and it may not return the comment immediately. Consider moving `setComments` into a `then` block after `createComment`.